src/box/lua/connpool.lua

```diff
@@ -196,19 +218,33 @@ local function filter(opts)
         instances = '?table',
         labels = '?table',
         roles = '?table',
+        sharding_roles = '?table',
```

Comment:
I would expect that all the `connpool.filter` filters are supported by `connpool.call`.

Reply:
Sounds. Fixed this strange inconsistency.

Reply:
The implementation looks OK, but, please, add a test case for `connpool.call` with the `sharding_roles` option.

Reply:
Added the test.